statistics/update.go

```diff
@@ -133,3 +136,60 @@ func (h *Handle) dumpTableStatDeltaToKV(id int64, delta variable.TableDelta) err
 	_, err = h.ctx.(sqlexec.SQLExecutor).Execute("commit")
 	return errors.Trace(err)
 }
+
+const (
+	// StatsOwnerKey is the stats owner path that is saved to etcd.
+	StatsOwnerKey = "/tidb/stats/owner"
+	// StatsPrompt is the prompt for stats owner manager.
+	StatsPrompt = "stats"
+)
+
+func needAnalyzeTable(tbl *Table, limit time.Duration) bool {
+	if tbl.ModifyCount == 0 {
+		return false
+	}
+	t := time.Unix(0, oracle.ExtractPhysical(tbl.Version)*int64(time.Millisecond))
+	if time.Now().Sub(t) < limit {
+		return false
+	}
+	for _, col := range tbl.Columns {
+		if len(col.Buckets) > 0 {
+			return false
+		}
+	}
+	for _, idx := range tbl.Indices {
+		if len(idx.Buckets) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// HandleAutoAnalyze analyze the newly created table or index.
+func (h *Handle) HandleAutoAnalyze(is infoschema.InfoSchema) error {
+	dbs := is.AllSchemas()
+	for _, db := range dbs {
+		for _, tbl := range db.Tables {
+			statsTbl := h.GetTableStats(tbl.ID)
+			if statsTbl.Pseudo || statsTbl.Count == 0 {
+				continue
+			}
+			tblName := db.Name.O + "." + tbl.Name.O
+			if needAnalyzeTable(statsTbl, 20*h.Lease) {
+				sql := fmt.Sprintf("analyze table %s", tblName)
+				log.Infof("[stats] auto analyze table %s now", tblName)
+				_, _, err := h.ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(h.ctx, sql)
+				return errors.Trace(err)
```

Comment:
1. Nope, currently every time we only analyze one table or index, to reduce the impact on the server. 
2. Only the log? It also analyzes the table.